stm/stm.ml

```diff
@@ -2510,9 +2510,7 @@ let known_state ~doc ?(redefine_qed=false) ~cache id =
                   match keep with
                   | VtDrop -> None
                   | VtKeep VtKeepAxiom ->
-                      let ctx = UState.empty in
-                      let fp = Future.from_val ([],ctx) in
-                      qed.fproof <- Some (fp, ref false); None
+                      qed.fproof <- Some (None, ref false); None
```

Comment:
>I'm OK with the opaque data type, but I don't get why you need to change the type of the Future thunk.

@gares Here we put some kind of dummy value
So either we expose `dummy : closed_proof_output` somewhere or we have to change the type of fproof